ArduCopter/avoidance.cpp

```diff
@@ -0,0 +1,20 @@
+#include "Copter.h"
+
+// check if proximity type Simple Avoidance should be enabled based on alt 
+void Copter::low_alt_avoidance()
+{   
+#if AC_AVOID_ENABLED == ENABLED
+    int32_t alt_cm;
+    if (!get_rangefinder_height_interpolated_cm(alt_cm)) {
+        // enable avoidance if we don't have a valid rangefinder reading
+        avoid.proximity_alt_avoidance_enable(true);
```

Comment:
This isn't sufficient; have a look at where we send rangefinder through to wpnav and circlenav - we're passing in the rangefinder state as well.

In particular, I'm worried that if someone misconfigurres things then they might avoid with an invalid rangefinder height (e.g. below-min-alt)

This call should be passing in the filtered rangefinder altitude, the rangefinder state and whether the rangefinder is enabled - the library should work out what to do with it.

Reply:
`get_rangefinder_height_interpolated_cm` is a robust method used by many modes directly at the Copter level. It returns flase if the range finder isn't healthy, or is disabled. If it returns true, then it populates the passed variable with the filtered value. We could pass all this information down to Avoidance lib, but I feel that since its purely a copter feature, its best we keep all the logic here.

Reply:
Thing is that the health is more nuanced than that.

At the moment the code will enable avoidance as soon as the vehicle goes below the rangefinder's minimum altitude (i.e. `Status::OutOfRangeLow`).  Given we want to *land* using this code, that's not good :-)